x-pack/plugins/security/server/audit/audit_service.ts

```diff
@@ -166,7 +166,7 @@ export class AuditService {
             session_id: sessionId,
             ...event.kibana,
           },
-          trace: { id: request.id },
+          trace: { id: request.opaqueId },
```

Comment:
@elastic/kibana-security `request.id` was renamed to `request.opaqueId`, but is now optional, as we're no longer generating a value when not provided by the client / when the configuration forbid its usage. 

Is this change fine, or should we fallback to `request.uuid` when `request.opaqueId` is empty?

Reply:
JFYI: Kibana sets tracing fields starting from https://github.com/elastic/kibana/pull/118466 The value from `meta` override them (sorry, I need to update the test suite name) 
https://github.com/elastic/kibana/blob/b288f90b3c929303b3b15433380cf751066c24f0/src/core/server/logging/layouts/json_layout.test.ts#L331
maybe audit logging should rely on the `trace` field supplied by the Kibana server and not overwrite them?

Reply:
yes, it wasn't. because it's not a bug fix

Reply:
> @elastic/kibana-security `request.id` was renamed to `request.opaqueId`, but is now optional, as we're no longer generating a value when not provided by the client / when the configuration forbid its usage.
> 
> Is this change fine, or should we fallback to `request.uuid` when `request.opaqueId` is empty?

I responded to the linked issue in https://github.com/elastic/kibana/issues/120124#issuecomment-1015877414.

I'm not on board with this change in general as it hamstrings auditing functionality. If we have to go through with this though, we *at least* need the ability to correlate audit records _within_ the Kibana audit log which originate from the same request. In that case it would be better to fall back to `request.uuid`.